test_sdist breaks with 6.3.0

### Checklist

- [X] This is a bug report and not [a different kind of issue](https://github.com/streamlink/streamlink/issues/new/choose)
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed bug reports](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22bug%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

6.3.0

### Description

Hey,
ArchLinux maintainer here. I'm having troubles building 6.3.0 due to a failing tests. Any hint what's wrong here?

Thank you!

### Debug log

```text
============================= test session starts ==============================
platform linux -- Python 3.11.5, pytest-7.4.3, pluggy-1.3.0
rootdir: /build/streamlink/src/streamlink-6.3.0
configfile: pyproject.toml
testpaths: build_backend, tests
plugins: asyncio-0.21.1, hypothesis-6.88.1, trio-0.8.0, requests-mock-1.11.0
asyncio: mode=Mode.STRICT
collected 6820 items

build_backend/test_build_backend.py .....                                [  0%]
build_backend/test_onbuild.py F.                                         [  0%]
tests/testutils/test_handshake.py ......                                 [  0%]
tests/utils/test_args.py ....................................            [  0%]
tests/utils/test_cache.py .                                              [  0%]
tests/utils/test_crypto.py ..                                            [  0%]
tests/utils/test_data.py ......                                          [  0%]
tests/utils/test_formatter.py ...                                        [  0%]
tests/utils/test_l10n.py ............................................... [  1%]
.                                                                        [  1%]
tests/utils/test_module.py ..                                            [  1%]
tests/utils/test_named_pipe.py .......ssssss                             [  1%]
tests/utils/test_parse.py ............                                   [  2%]
tests/utils/test_path.py ...........                                     [  2%]
tests/utils/test_processoutput.py ........                               [  2%]
tests/utils/test_random.py .........                                     [  2%]
tests/utils/test_times.py .............................................. [  3%]
........................................................................ [  4%]
........................................                                 [  5%]
tests/utils/test_url.py ...................                              [  5%]
tests/test_api_http_session.py ............................              [  5%]
tests/test_api_validate.py ............................................. [  6%]
........................................................................ [  7%]
.........................................................                [  8%]
tests/test_api_websocket.py ..................                           [  8%]
tests/test_buffers.py .................                                  [  9%]
tests/test_cache.py .....................                                [  9%]
tests/test_logger.py ........................................            [ 10%]
tests/test_options.py ....................                               [ 10%]
tests/test_plugin.py ............................................        [ 11%]
tests/test_plugin_userinput.py ....                                      [ 11%]
tests/test_session.py ...........................................        [ 11%]
tests/test_streamlink_api.py .....                                       [ 12%]
tests/webbrowser/test_chromium.py ..................                     [ 12%]
tests/webbrowser/test_webbrowser.py ............                         [ 12%]
tests/webbrowser/cdp/test_client.py .................................... [ 13%]
..........                                                               [ 13%]
tests/webbrowser/cdp/test_connection.py ..........................       [ 13%]
tests/stream/test_ffmpegmux.py ......................................... [ 14%]
                                                                         [ 14%]
tests/stream/test_file.py ..                                             [ 14%]
tests/stream/test_segmented.py .                                         [ 14%]
tests/stream/test_stream_json.py ........                                [ 14%]
tests/stream/test_stream_to_url.py ........                              [ 14%]
tests/stream/test_stream_wrappers.py .                                   [ 14%]
tests/stream/dash/test_dash.py ...........................               [ 15%]
tests/stream/dash/test_manifest.py ..................................... [ 15%]
..                                                                       [ 15%]
tests/stream/hls/test_hls.py ........................................... [ 16%]
.......                                                                  [ 16%]
tests/stream/hls/test_hls_filtered.py ......                             [ 16%]
tests/stream/hls/test_m3u8.py .......................................... [ 17%]
........................                                                 [ 17%]
tests/test_plugins.py .................................................. [ 18%]
........................................................................ [ 19%]
........................................................................ [ 20%]
........................................................................ [ 21%]
........................................................................ [ 23%]
........................................................................ [ 24%]
........................................................................ [ 25%]
........................................................................ [ 26%]
........................................................................ [ 27%]
........................................................................ [ 28%]
........................................................................ [ 29%]
........................................................................ [ 31%]
........................................................................ [ 32%]
........................................................................ [ 33%]
........................................................................ [ 34%]
........................................................................ [ 35%]
........................................................................ [ 36%]
........................................................................ [ 38%]
........................................................................ [ 39%]
........................................................................ [ 40%]
........................................................................ [ 41%]
........................................................................ [ 42%]
........................................................................ [ 43%]
........................................................................ [ 44%]
........................................................................ [ 46%]
........................................................................ [ 47%]
........................................................................ [ 48%]
........................................................................ [ 49%]
........................................................................ [ 50%]
........................................................................ [ 51%]
........................................................................ [ 52%]
.                                                                        [ 52%]
tests/plugins/test_abematv.py ..................                         [ 53%]
tests/plugins/test_adultswim.py ............                             [ 53%]
tests/plugins/test_afreeca.py ...............                            [ 53%]
tests/plugins/test_albavision.py ....................................... [ 54%]
.........                                                                [ 54%]
tests/plugins/test_aloula.py ..................................          [ 55%]
tests/plugins/test_app17.py .........                                    [ 55%]
tests/plugins/test_ard_live.py .........                                 [ 55%]
tests/plugins/test_ard_mediathek.py ..................                   [ 55%]
tests/plugins/test_artetv.py .......................                     [ 55%]
tests/plugins/test_atpchallenger.py .........                            [ 56%]
tests/plugins/test_atresplayer.py ..............                         [ 56%]
tests/plugins/test_bbciplayer.py ..........                              [ 56%]
tests/plugins/test_bfmtv.py ...................                          [ 56%]
tests/plugins/test_bigo.py .......................                       [ 57%]
tests/plugins/test_bilibili.py .......                                   [ 57%]
tests/plugins/test_blazetv.py ...............                            [ 57%]
tests/plugins/test_bloomberg.py .................                        [ 57%]
tests/plugins/test_booyah.py ..........................                  [ 58%]
tests/plugins/test_brightcove.py ........                                [ 58%]
tests/plugins/test_btv.py .........                                      [ 58%]
tests/plugins/test_cbsnews.py ....................                       [ 58%]
tests/plugins/test_cdnbg.py ................................             [ 59%]
tests/plugins/test_ceskatelevize.py ...................                  [ 59%]
tests/plugins/test_cinergroup.py .................                       [ 59%]
tests/plugins/test_clubbingtv.py ..........                              [ 60%]
tests/plugins/test_cmmedia.py ................                           [ 60%]
tests/plugins/test_cnews.py ..........                                   [ 60%]
tests/plugins/test_crunchyroll.py .................................      [ 60%]
tests/plugins/test_dailymotion.py .............                          [ 61%]
tests/plugins/test_dash.py ............................................. [ 61%]
..........................                                               [ 62%]
tests/plugins/test_delfi.py ...........                                  [ 62%]
tests/plugins/test_deutschewelle.py ...............                      [ 62%]
tests/plugins/test_dlive.py ......................                       [ 63%]
tests/plugins/test_dogan.py .........................                    [ 63%]
tests/plugins/test_dogus.py ..............                               [ 63%]
tests/plugins/test_drdk.py ............                                  [ 63%]
tests/plugins/test_earthcam.py ........                                  [ 64%]
tests/plugins/test_euronews.py .....................                     [ 64%]
tests/plugins/test_facebook.py ............                              [ 64%]
tests/plugins/test_filmon.py ..........................................  [ 65%]
tests/plugins/test_foxtr.py .......                                      [ 65%]
tests/plugins/test_galatasaraytv.py ..........                           [ 65%]
tests/plugins/test_goltelevision.py ..............                       [ 65%]
tests/plugins/test_goodgame.py .................................         [ 66%]
tests/plugins/test_googledrive.py .......                                [ 66%]
tests/plugins/test_gulli.py ..............                               [ 66%]
tests/plugins/test_hiplayer.py ............                              [ 66%]
tests/plugins/test_hls.py .............................................. [ 67%]
................................................                         [ 68%]
tests/plugins/test_http.py .......................                       [ 68%]
tests/plugins/test_htv.py ........................                       [ 69%]
tests/plugins/test_huajiao.py ........                                   [ 69%]
tests/plugins/test_huya.py ............                                  [ 69%]
tests/plugins/test_idf1.py ............                                  [ 69%]
tests/plugins/test_indihometv.py .........                               [ 69%]
tests/plugins/test_invintus.py ...........                               [ 69%]
tests/plugins/test_kugou.py ............                                 [ 70%]
tests/plugins/test_linelive.py ...........                               [ 70%]
tests/plugins/test_livestream.py .............................           [ 70%]
tests/plugins/test_lnk.py ...............                                [ 70%]
tests/plugins/test_lrt.py ..............                                 [ 71%]
tests/plugins/test_ltv_lsm_lv.py ...................                     [ 71%]
tests/plugins/test_mdstrm.py ...............                             [ 71%]
tests/plugins/test_mediaklikk.py ..................                      [ 71%]
tests/plugins/test_mediavitrina.py ..................................... [ 72%]
..                                                                       [ 72%]
tests/plugins/test_mildom.py ..........                                  [ 72%]
tests/plugins/test_mitele.py ................                            [ 73%]
tests/plugins/test_mixcloud.py ....................                      [ 73%]
tests/plugins/test_mjunoon.py ...........                                [ 73%]
tests/plugins/test_mrtmk.py ...........                                  [ 73%]
tests/plugins/test_n13tv.py ..................                           [ 73%]
tests/plugins/test_nhkworld.py .......                                   [ 74%]
tests/plugins/test_nicolive.py ..............                            [ 74%]
tests/plugins/test_nimotv.py ...........                                 [ 74%]
tests/plugins/test_nos.py ................                               [ 74%]
tests/plugins/test_nownews.py ............                               [ 74%]
tests/plugins/test_nrk.py ...................                            [ 75%]
tests/plugins/test_okru.py .............                                 [ 75%]
tests/plugins/test_olympicchannel.py ..................                  [ 75%]
tests/plugins/test_oneplusone.py ...............                         [ 75%]
tests/plugins/test_onetv.py .....................                        [ 76%]
tests/plugins/test_openrectv.py .........                                [ 76%]
tests/plugins/test_pandalive.py ........                                 [ 76%]
tests/plugins/test_piaulizaportal.py ............                        [ 76%]
tests/plugins/test_picarto.py ................                           [ 77%]
tests/plugins/test_piczel.py ..........                                  [ 77%]
tests/plugins/test_pixiv.py .........                                    [ 77%]
tests/plugins/test_pluto.py ............................................ [ 78%]
..........                                                               [ 78%]
tests/plugins/test_pluzz.py .................                            [ 78%]
tests/plugins/test_qq.py ............                                    [ 78%]
tests/plugins/test_radiko.py ............                                [ 78%]
tests/plugins/test_radionet.py ..................                        [ 79%]
tests/plugins/test_raiplay.py .............                              [ 79%]
tests/plugins/test_reuters.py ............                               [ 79%]
tests/plugins/test_rtbf.py ............                                  [ 79%]
tests/plugins/test_rtpa.py ........                                      [ 79%]
tests/plugins/test_rtpplay.py ..................                         [ 80%]
tests/plugins/test_rtve.py ..................                            [ 80%]
tests/plugins/test_rtvs.py ................                              [ 80%]
tests/plugins/test_ruv.py ....................                           [ 81%]
tests/plugins/test_sbscokr.py ..............                             [ 81%]
tests/plugins/test_showroom.py ............                              [ 81%]
tests/plugins/test_sportal.py .........                                  [ 81%]
tests/plugins/test_sportschau.py ........                                [ 81%]
tests/plugins/test_ssh101.py ...........                                 [ 81%]
tests/plugins/test_stadium.py ...........                                [ 82%]
tests/plugins/test_steam.py .............                                [ 82%]
tests/plugins/test_streamable.py .......                                 [ 82%]
tests/plugins/test_streann.py ................                           [ 82%]
tests/plugins/test_stv.py ........                                       [ 82%]
tests/plugins/test_svtplay.py .................                          [ 83%]
tests/plugins/test_swisstxt.py ..................                        [ 83%]
tests/plugins/test_telefe.py ..............                              [ 83%]
tests/plugins/test_telemadrid.py .........                               [ 83%]
tests/plugins/test_tf1.py ............................                   [ 84%]
tests/plugins/test_trovo.py ..............................               [ 84%]
tests/plugins/test_turkuvaz.py ................................          [ 85%]
tests/plugins/test_tv360.py .......                                      [ 85%]
tests/plugins/test_tv3cat.py ..............                              [ 85%]
tests/plugins/test_tv4play.py ..........                                 [ 85%]
tests/plugins/test_tv5monde.py .............                             [ 85%]
tests/plugins/test_tv8.py .......                                        [ 85%]
tests/plugins/test_tv999.py ............                                 [ 86%]
tests/plugins/test_tvibo.py .........                                    [ 86%]
tests/plugins/test_tviplayer.py ..........                               [ 86%]
tests/plugins/test_tvp.py ....................................           [ 86%]
tests/plugins/test_tvrby.py ............                                 [ 87%]
tests/plugins/test_tvrplus.py ...............                            [ 87%]
tests/plugins/test_tvtoya.py ..............                              [ 87%]
tests/plugins/test_twitcasting.py .......                                [ 87%]
tests/plugins/test_twitch.py ........................................... [ 88%]
...............                                                          [ 88%]
tests/plugins/test_ustreamtv.py ...........................              [ 89%]
tests/plugins/test_ustvnow.py .........                                  [ 89%]
tests/plugins/test_vidio.py ...........                                  [ 89%]
tests/plugins/test_vimeo.py .................                            [ 89%]
tests/plugins/test_vinhlongtv.py ...........                             [ 89%]
tests/plugins/test_vk.py ..............................                  [ 90%]
tests/plugins/test_vkplay.py ...........                                 [ 90%]
tests/plugins/test_vtvgo.py ..................                           [ 90%]
tests/plugins/test_wasd.py ...........                                   [ 91%]
tests/plugins/test_webtv.py .................                            [ 91%]
tests/plugins/test_welt.py ...................                           [ 91%]
tests/plugins/test_wwenetwork.py ..........                              [ 91%]
tests/plugins/test_youtube.py .......................................... [ 92%]
.................                                                        [ 92%]
tests/plugins/test_yupptv.py ........                                    [ 92%]
tests/plugins/test_zattoo.py ......................................      [ 93%]
tests/plugins/test_zdf_mediathek.py .................                    [ 93%]
tests/plugins/test_zeenews.py ........                                   [ 93%]
tests/plugins/test_zengatv.py ............                               [ 94%]
tests/plugins/test_zhanqi.py .......                                     [ 94%]
tests/cli/test_argparser.py ...............................              [ 94%]
tests/cli/test_cmdline.py ....                                           [ 94%]
tests/cli/test_cmdline_title.py ....sssss                                [ 94%]
tests/cli/test_console.py ..............                                 [ 95%]
tests/cli/test_main.py ................................................s [ 95%]
ss....                                                                   [ 95%]
tests/cli/test_main_formatter.py .........                               [ 96%]
tests/cli/test_main_setup_config_args.py ............                    [ 96%]
tests/cli/test_plugin_args_and_options.py ...                            [ 96%]
tests/cli/test_streamrunner.py .....................                     [ 96%]
tests/cli/output/test_file.py ....s                                      [ 96%]
tests/cli/output/test_player.py ........s.......s.ss.....s.s............ [ 97%]
                                                                         [ 97%]
tests/cli/utils/test_formatter.py ...                                    [ 97%]
tests/cli/utils/test_path.py .................................s.s.s...s  [ 98%]
tests/cli/utils/test_player.py sssss.....                                [ 98%]
tests/cli/utils/test_progress.py ....................................... [ 98%]
......................................................s.                 [ 99%]
tests/cli/utils/test_versioncheck.py ...............                     [100%]

=================================== FAILURES ===================================
__________________________ test_sdist[is_source=True] __________________________

build = PosixPath('/tmp/pytest-of-builduser/pytest-0/test_sdist_is_source_True_0')

    @pytest.mark.parametrize("build", [pytest.param({"is_source": True}, id="is_source=True")], indirect=True)
    def test_sdist(build: Path):
        assert re.search(
            r"^(\s*)# (\"versioningit\b.+?\",).*$",
            (build / "pyproject.toml").read_text(encoding="utf-8"),
            re.MULTILINE,
        ), "versioningit is not a build-requirement"
>       assert re.search(
            r"^(\s*)(version=\"1\.2\.3\+fake\",).*$",
            (build / "setup.py").read_text(encoding="utf-8"),
            re.MULTILINE,
        ), "setup() call defines a static version string"
E       AssertionError: setup() call defines a static version string
E       assert None
E        +  where None = <function search at 0x7f922c01bb00>('^(\\s*)(version=\\"1\\.2\\.3\\+fake\\",).*$', '#!/usr/bin/env python\nfrom os import path\nfrom sys import argv, exit, version_info\nfrom textwrap import dedent\n\n...t_cmdclasses(),\n        entry_points=entry_points,\n        data_files=data_files,\n        version="6.3.0",\n    )\n', re.MULTILINE)
E        +    where <function search at 0x7f922c01bb00> = re.search
E        +    and   '#!/usr/bin/env python\nfrom os import path\nfrom sys import argv, exit, version_info\nfrom textwrap import dedent\n\n...t_cmdclasses(),\n        entry_points=entry_points,\n        data_files=data_files,\n        version="6.3.0",\n    )\n' = <bound method Path.read_text of PosixPath('/tmp/pytest-of-builduser/pytest-0/test_sdist_is_source_True_0/setup.py')>(encoding='utf-8')
E        +      where <bound method Path.read_text of PosixPath('/tmp/pytest-of-builduser/pytest-0/test_sdist_is_source_True_0/setup.py')> = (PosixPath('/tmp/pytest-of-builduser/pytest-0/test_sdist_is_source_True_0') / 'setup.py').read_text
E        +    and   re.MULTILINE = re.MULTILINE

build_backend/test_onbuild.py:46: AssertionError
=========================== short test summary info ============================
FAILED build_backend/test_onbuild.py::test_sdist[is_source=True] - AssertionE...
================= 1 failed, 6228 passed, 31 skipped in 16.55s ==================
```

--- a/build_backend/test_onbuild.py
+++ b/build_backend/test_onbuild.py
@@ -44,7 +44,10 @@ def test_sdist(build: Path):
         re.MULTILINE,
     ), "versioningit is not a build-requirement"
     assert re.search(
-        r"^(\s*)(version=\"1\.2\.3\+fake\",).*$",
+        # Check for any version string, and not just the fake one,
+        # because tests can be run from the sdist where the version string was already set.
+        # The onbuild hook does only replace the template in `setup.py`.
+        r"^(\s*)(version=\"[^\"]+\",).*$",
         (build / "setup.py").read_text(encoding="utf-8"),
         re.MULTILINE,
     ), "setup() call defines a static version string"

